Add a MetaICP class

v4hn opened the pull request:
This is quite similar in spirit to IncrementalICP, but matches new clouds to
all previously registered clouds instead of only the last one. This is
computationally more expensive, but provides a more robust matching when
successive clouds do not alway cover nearby parts of the environment.

As this (just as IncrementalICP) is a standard method to apply ICP in
practice, I expected to find this functionality in PCL. As it wasn't there
yet, I wrote this class.

taketwo commented:
This is not limited to ICP, any registration algorithm will work, right?

v4hn commented:
Yes. I was mirroring the IncrementalICP class (which also is not limited to
ICP), so I called it MetaICP. This is also the term for this method in
my working group.

jspricke commented:
I would propose to rename both classes to *Registration.

taketwo commented:
+1 for renaming to *Registration.
Also, could you adapt to PCL style guide (line breaks before curly braces, no spaces inside braces)?

taketwo commented on registration/include/pcl/registration/meta_icp.h:
```diff
@@ -0,0 +1,125 @@
+        bool
+        registerCloud (const PointCloudConstPtr& cloud, const Matrix4& delta_estimate = Matrix4::Identity ());
+
+        /** \brief Get estimated transform */
```
... of the last registered cloud?

taketwo commented on registration/include/pcl/registration/meta_icp.h:
```diff
@@ -0,0 +1,125 @@
+      protected:
+
+        /** \brief registered meta point cloud */
+        PointCloudPtr full_cloud_;
```
Would it make sense to provide an constant accessor to this?

taketwo commented on registration/include/pcl/registration/meta_icp.h:
```diff
@@ -0,0 +1,125 @@
+      *   write_cloud (*tmp);
+      * }
+      * \endcode
+      *
```
Maybe add a note that a meta cloud is being accumulated? I think it's important for users to be aware of so that there is no surprise where all the memory went.

v4hn commented:
I renamed the class to MetaRegistration and adjusted the interfaces as proposed.
As this request is not about IncrementalICP, I will not add changes for that here.
You're welcome to add a patch for this class yourself.
